### PR TITLE
Milestone_2

### PR DESCRIPTION
**Pull request by Farida-Hamid**
### Create a database based on a diagram:
- Created a file named `schema_based_on_diagram.sql` where we implement the database from the diagram.
- Joined tables from many-to-many relationships.
- Add the FK indexes.

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    patients_id INT,
+    status varchar(255),
+    FOREIGN KEY(patients_id) REFERENCES patients(id)
+ );
+ 
+CREATE TABLE treatments(
+    id serial PRIMARY KEY, 
+    type VARCHAR(50),
+    name VARCHAR(50)
+    );
```
Comment (zieeco) [inline schema_based_on_diagram.sql:19]: - Let's create the "join table" for the many-to-many relationships between `medical_histories` and `treatments` 
Below is an example:
  ```sql
    CREATE TABLE medical_histories_has_treatments (
      medical_history_id int refrences medical_histories(id),
      treatment_id int refrences treatments(id),
      );
  ```
---
- Let's also ensure that all FOREIGN KEYS have indexes created e.g:

`CREATE INDEX ON medical_histories (patient_id);`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Team,


Great job so far 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- Nice schema.✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Farida-Hamid) [reply schema_based_on_diagram.sql:19]: We did include the instruction:
`CREATE INDEX ON medical_histories (patient_id);` in line 21.

Comment (Nemwel-Boniface) [inline schema_based_on_diagram.sql:19]: - [ ] As requested by the previous reviewer @zieeco  and according to the project requirements, you are required to create a join table for your project. He was kind enough to point out to your team exactly which tables need the join table. Kindly create the join table as was previously requested.

image

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+CREATE INDEX treatment_histories ON medical_histories(id);
```
Comment (Nemwel-Boniface) [inline schema_based_on_diagram.sql]: - [ ] I have seen that you have one index created at line 21 as you @Farida-Hamid  have said. However, according to the project requirements, you are required to create indexes for all the foreign keys for all the tables in your project. What this means is that you need to have a total of six indexes, this includes the 5 primary tables plus one for the join table. I hope this is much clearer now for your team.

image

**Review by Nemwel-Boniface — CHANGES_REQUESTED** [review]
## Hello @Farida-Hamid, @kessio  and @AbigiyaTY       :wave: , 
I'm Nemwel your reviewer for this PR :man_beard: 

Great job addressing some of the changes that were requested by a previous reviewer :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️  


## ♻️ Required Changes ♻️

_Check the comments under the review._

## [Optional] suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding! :partying_face: 👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    invoice_id INT REFERENCES invoices(id), 
+    treatment_id INT,
+    FOREIGN KEY (treatment_id) REFERENCES treatments(id),
+    FOREIGN KEY (invoice_id) REFERENCES invoices(id)    
+);
+
+CREATE TABLE medical_histories_has_treatments (
+    medical_history_id int refrences medical_histories(id),
+    treatment_id int refrences treatments(id),
+    );
```
Comment (gemmen29) [inline schema_based_on_diagram.sql]: - [ ] Kindly add indexes for those two foreign keys(`medical_history_id`, `treatment_id`) for the table `medical_histories_has_treatments` as requested by the previous reviewers.

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯  
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codecaiine — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Goo database setup :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**